rdt-open-reader/src/main/java/io/ona/rdt/interactor/PatientRegisterFragmentInteractor.java

```diff
@@ -136,18 +141,20 @@ private void populateApproxDOB(JSONArray fields) throws JSONException {
         }
     }
 
-    private void closeRDTId(org.smartregister.domain.db.Event dbEvent) {
-        try {
-            org.smartregister.domain.db.Obs rdtIdObs = dbEvent.findObs(null,
-                    false,
-                    RDTApplication.getInstance().getStepStateConfiguration().getStepStateObj().optString(RDT_ID_KEY));
-
-            if (rdtIdObs != null) {
-                String rdtId = rdtIdObs.getValue() == null ? "" : rdtIdObs.getValue().toString();
+    private void closeIDs(org.smartregister.domain.db.Event dbEvent) {
+        // close rdt id
+        org.smartregister.domain.db.Obs idObs = dbEvent.findObs(null, false, RDT_ID);
+        if (idObs != null) {
+            String rdtId = idObs.getValue() == null ? "" : idObs.getValue().toString();
+            RDTApplication.getInstance().getContext().getUniqueIdRepository().close(rdtId);
+        }
+        // close respiratory sample id
+        if (isCovidApp()) {
```

Comment:
does this mean we're assuming every time you do a Covid RDT you also do a respiratory sample?